internal/buildjob/buildjob.go

```diff
@@ -187,78 +187,155 @@ func (j *Job) generateBuildOptions(ctx context.Context, cib *v1alpha1.ContainerI
+			configuredRegistries[apiReg.Server] = config.Registry{
+				Host:   apiReg.Server,
+				NonSSL: apiReg.NonSSL,
+			}
 		}
+	}
 
-		configs = append(configs, conf)
+	registryConfigs = []config.Registry{}
+	for _, v := range configuredRegistries {
+		registryConfigs = append(registryConfigs, v)
 	}
 
-	return configs, nil
+	return registryConfigs, nil
 }
 
-func (j *Job) getDockerAuthFromFS(host string) (string, string, error) {
+func (j *Job) getDockerAuthsFromFS() (credentials.AuthConfigs, error) {
 	if _, err := os.Stat(config.DynamicCredentialsFilepath); os.IsNotExist(err) {
-		return "", "", errors.Wrap(err, "filesystem docker credentials missing")
+		return nil, errors.Wrap(err, "filesystem docker credentials missing")
 	}
 
 	input, err := ioutil.ReadFile(config.DynamicCredentialsFilepath)
 	if err != nil {
-		return "", "", errors.Wrap(err, "cannot read docker credentials")
+		return nil, errors.Wrap(err, "cannot read docker credentials")
 	}
 
-	username, password, err := credentials.ExtractDockerAuth(input, host)
+	authConfigs, err := credentials.ExtractAuthConfigs(input)
 	if err != nil {
-		return "", "", errors.Wrapf(err, "cannot process filesystem docker auth for host %q", host)
+		return nil, errors.Wrapf(err, "Failed to read docker auth from filesystem")
 	}
 
-	return username, password, nil
+	return authConfigs, nil
 }
 
-func (j *Job) getDockerAuthFromSecret(ctx context.Context, host, name, namespace string) (string, string, error) {
-	secret, err := j.clientk8s.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+func (j *Job) getDockerAuthsFromSecret(ctx context.Context, secretName string, secretNamespace string) (credentials.AuthConfigs, error) {
+	secret, err := j.clientk8s.CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
 	if err != nil {
-		return "", "", errors.Wrap(err, "cannot find registry auth secret")
+		return nil, errors.Wrap(err, "cannot find registry auth secret")
 	}
 	if secret.Type != corev1.SecretTypeDockerConfigJson {
-		return "", "", fmt.Errorf("registry auth secret must be %v, not %v", corev1.SecretTypeDockerConfigJson, secret.Type)
+		return nil, fmt.Errorf("registry auth secret must be %v, not %v", corev1.SecretTypeDockerConfigJson, secret.Type)
+	}
+
+	authConfigs, err := credentials.ExtractAuthConfigs(secret.Data[corev1.DockerConfigJsonKey])
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read docker auth from secret")
 	}
 
-	username, password, err := credentials.ExtractDockerAuth(secret.Data[corev1.DockerConfigJsonKey], host)
+	return authConfigs, nil
+}
+
+func (j *Job) getBasicAuthForHost(authConfigs credentials.AuthConfigs, host string) (string, string, error) {
+	username, password, err := credentials.ExtractBasicAuthForHost(authConfigs, host)
 	if err != nil {
-		return "", "", errors.Wrap(err, "cannot process docker auth from secret")
+		return "", "", errors.Wrapf(err, "Failed to extract docker auth for specified host (%q)", host)
```

Comment:
The parentheses are stylistic and purely meant to provide visual clarity to help pick out the host string from the line of rendered text

Reply:
Right but (abc) vs ("abc") could just be `specified host, "host"` or just `specificed host, HOST`
normally it would be context on the log rather than text.